runtime/drivers/bigquery/warehouse.go

```diff
@@ -242,19 +242,34 @@ func (f *fileIterator) Next() ([]string, error) {
 
 	ticker := time.NewTicker(time.Second * 5)
 	defer ticker.Stop()
+
+	limitExceeded := make(chan struct{})
+	defer close(limitExceeded)
+
+	go func() {
+		for {
+			select {
+			case <-limitExceeded:
```

Comment:
I think this is an issue that same channel is received by this goroutine as well as the main goroutine. One of these will not receive it.
It is also not idiomatic to send and receive on a channel from the same goroutine. 
Also I think in cases when limits are not exceeded(or main goroutine receives the value) this goroutine will leak waiting on both channels (unless `ticker.Stop` closes the channel).
I think we can override the parent context with a cancellable context and cancel the context on limit exceeds. This goroutine can listen on `ctx.Done()` and exit.

Reply:
thanks fixed.